### DISCUSSION TITLE
Is it appropriate to report "code is unreachable" when not TYPE_CHECKING?

### DISCUSSION THREAD

**Discussion by dnaaun** [microsoft/pyright#2330] [Q&A] [answered]
Consider this:

```py
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    pass
else:
    pass # Pyright complains "Code is unreachable"
```

Given that `TYPE_CHECKING` is basically saying, "trust me type checker, I know this better than you", is it appropriate to report that code is unreachable in this case?

**Comment by erictraut** [comment, marked as answer]
The code in this area is not analyzed because it is unreachable from the perspective of the code flow engine. I'd argue that it's important for the user to understand that the code in this section is not being analyzed. We've talked about perhaps changing the wording from "Code is unreachable" to something else, but I haven't come up with any phrasing that isn't verbose or more confusing. Let me know if you have suggestions.

**Reply by dnaaun** [reply]
> I'd argue that it's important for the user ...

I thought this was an oversight, but I see you've given it thought.

> I haven't come up with any phrasing that isn't verbose ...

I can't think of any such phrasing either.
